### ISSUE TITLE
Losses turn nan due to negative values of boxes

### ISSUE THREAD

**Issue by vipulbjj** [qfgaohao/pytorch-ssd#112]
target_transform causes box to have negative values which when passed in assign_priors function goes under a log function causing nan values. How to solve this?

**Comment by ishangupta3** [comment]
Will this effect the results ?

**Comment by vipulbjj** [comment]
I got the issue. Actually height or width of boxes is getting to 0 at some places due to minor errors in the dataset and we're dividing by it in convert_boxes_to_locations. maybe adding a small value(like epsilon) while dividing might sort it

**Comment by ishang3** [comment]
Interesting; in the mean  i was able to get training done on a custom dataset using this repo:
https://github.com/midasklr/SSD.Pytorch

Please do let me know if you are able to get around that error; I am using a custom dataset and i am positive none of my bounding boxes have a height or width of 0.

**Comment by vipulbjj** [comment]
It started to work, just make sure none of your XMin is equal to XMax and same for Y.

**Comment by ishangupta3** [comment]
Will this mean that we are feeding incorrect data for training. It seems that their transforms must be removing some of the bounding boxes out of the frame. 

Where are you checking these conditions that the XMin is not equal to XMax in the code?

Will it be somewhere here:

class MatchPrior(object):
    def __init__(self, center_form_priors, center_variance, size_variance, iou_threshold):
        self.center_form_priors = center_form_priors
        self.corner_form_priors = box_utils.center_form_to_corner_form(center_form_priors)
        self.center_variance = center_variance
        self.size_variance = size_variance
        self.iou_threshold = iou_threshold

    def __call__(self, gt_boxes, gt_labels):
        if type(gt_boxes) is np.ndarray:
            gt_boxes = torch.from_numpy(gt_boxes)
        if type(gt_labels) is np.ndarray:
            gt_labels = torch.from_numpy(gt_labels)
        boxes, labels = box_utils.assign_priors(gt_boxes, gt_labels,
                                                self.corner_form_priors, self.iou_threshold)
        boxes = box_utils.corner_form_to_center_form(boxes)
        locations = box_utils.convert_boxes_to_locations(boxes, self.center_form_priors, self.center_variance, self.size_variance)
        return locations, labels

Thanks for the previous tips.

**Comment by manaclan** [comment]
add these after line 108 in vision/datasets/voc_dataset.py help me solve the problem:
`                if x1 > x2:
                    x1, x2 = x2, x1
                if y1 > y2:
                    y1, y2 = y2, y1`
